northd/northd.c

```diff
@@ -8644,24 +8644,38 @@ build_lswitch_arp_nd_responder_known_ips(struct ovn_port *op,
                 }
             }
         }
-
-        if (op->peer) {
-            const char *arp_proxy = smap_get(&op->nbsp->options,"arp_proxy");
-
+        const char *arp_proxy = smap_get(&op->nbsp->options,"arp_proxy");
+        if (arp_proxy) {
             struct lport_addresses proxy_arp_addrs;
-            int i = 0;
+            int i, ofs = 0;
+            /* Either takes "MAC IP1 IP2" or "IP1 IP2" */
+            if (!extract_addresses(arp_proxy, &proxy_arp_addrs, &ofs) &&
+                !extract_ip_addresses(arp_proxy, &proxy_arp_addrs)) {
+                static struct vlog_rate_limit rl = VLOG_RATE_LIMIT_INIT(1, 5);
+                VLOG_WARN_RL(&rl, "Invalid arp_proxy option: '%s' at lsp '%s'",
+                    arp_proxy, op->nbsp->name);
+                return;
+            }
+
+            /* Select the mac address to answer the proxy ARP/NDP */
+            char* ea_s = NULL;
+            if (strlen(proxy_arp_addrs.ea_s)) {
```

Comment:
We cannot use that since `.ea` is a string not a `eth_addr` struct

If there is not a mac address is initialized [here](https://github.com/ovn-org/ovn/blob/main/lib/ovn-util.c#L134-L138).

```c
if (!ovs_scan_len(buf, &buf_index, ETH_ADDR_SCAN_FMT,
                          ETH_ADDR_SCAN_ARGS(laddrs->ea))) {
            laddrs->ea = eth_addr_zero;
            *ofs = 0;
            return false;
        }
```
So since it's an empty string we cannot only use strlen